pandas/indexes/multi.py

```diff
@@ -864,9 +862,9 @@ def from_arrays(cls, arrays, sortorder=None, names=None):
             if len(arrays[i]) != len(arrays[i - 1]):
                 raise ValueError('all arrays must be same length')
 
-        cats = [Categorical.from_array(arr, ordered=True) for arr in arrays]
```

Comment:
There are just a few places: internals of `concat` and `unstack`, `LegacyTable` in pytables, and `panel_index`. I can replace them all with `_factorize` - it's pretty straightforward. It wouldn't automatically fix `concat` and `unstack` (that's why I opened separate issues for them) but wouldn't hurt either.

If we replace them, what should be done to the definition of `Categorical.from_array`? Remove completely or rather add a deprecation warning?